torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -439,14 +439,13 @@ class FullyShardedDataParallel(nn.Module):
         previously initialized optimizers.
 
     .. warning::
-        ``module`` should be already placed on the destination device or
-        device is set properly using ``torch.cuda.set_device(device_id)``.
-        Alternatively, if module is input on CPU, ``device_id`` argument can
-        be specified to move module to CUDA device during initialization.
-        FSDP will get compute device from module first, if module device
-        is CPU, FSDP will then get compute device from current device. Note
-        that if ``device_id`` is specified, compute device will always match
-        ``device_id``.
+        If the destination CUDA device has ID ``dev_id``, either (1)
+        ``module`` should already be placed on that device, (2) the device
```

Comment:
I think doing (2) is actually different from doing (1) or (3).

If we do (2) and module is on CPU, sharding/init will still run on CPU. But for (1) or (3), it would run on GPU.